storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -139,12 +139,12 @@
                   (.add local pair) 
 
                   ;;Using java objects directly to avoid performance issues in java code
-                  (let [node+port (get @task->node+port task)]
-                    (when (not (.get remoteMap node+port))
-                      (.put remoteMap node+port (ArrayList.)))
-                    (let [remote (.get remoteMap node+port)]
+                  (let []
+                    (when (not (.get remoteMap task))
+                      (.put remoteMap task (ArrayList.)))
+                    (let [remote (.get remoteMap task)]
                       (.add remote (TaskMessage. task (.serialize serializer tuple)))
-                     )))) 
+                    ))))
```

Comment:
@d2r 
Think about situation like 

* tuples which consist of 
  * 3 tuples corresponding to task 1 (worker A) -- (A)
  * 5 tuples corresponding to task 2 (worker A) -- (B)
  * 2 tuples corresponding to task 3 (worker B) -- (C)

We can ensure we send tuples to proper worker since we check task->node+port with read lock.
But if we don't re-group messages by worker, we'll send tuples batched by task, not worker.
It means, though we can send 10 tuples at a moment, we send batched tuples (A), (B) to worker A separately.